File: PHPMovieDB/db/MovieAccessor.php
<?php

require_once 'ConnectionManager.php';
require_once (__DIR__ . '/../entity/Movie.php');

class MovieAccessor {

    private $getByIDStatementString = "select * from MOVIES where movieID = :movieID";
    private $deleteStatementString = "delete from Movies where movieID = :movieID";
    private $insertStatementString = "insert INTO Movies values (:movieID, :title, :releaseYear, :length, :rating)";
    private $updateStatementString = "update Movies set movieID = :movieID, title = :title, releaseYear = :releaseYear, rating = :rating where movieID = :movieID";
    private $conn = NULL;
    private $getByIDStatement = NULL;
    private $deleteStatement = NULL;
    private $insertStatement = NULL;
    private $updateStatement = NULL;

    // Constructor will throw exception if there is a problem with ConnectionManager, or with the prepared statements.
    public function __construct() {
        $cm = new ConnectionManager();

        $this->conn = $cm->connect_db();
        if (is_null($this->conn)) {
            throw new Exception("no connection");
        }
        $this->getByIDStatement = $this->conn->prepare($this->getByIDStatementString);
        if (is_null($this->getByIDStatement)) {
            throw new Exception("bad statement: '" . $this->getAllStatementString . "'");
        }

        $this->deleteStatement = $this->conn->prepare($this->deleteStatementString);
        if (is_null($this->deleteStatement)) {
            throw new Exception("bad statement: '" . $this->deleteStatementString . "'");
        }

        $this->insertStatement = $this->conn->prepare($this->insertStatementString);
        if (is_null($this->insertStatement)) {
            throw new Exception("bad statement: '" . $this->getAllStatementString . "'");
        }

        $this->updateStatement = $this->conn->prepare($this->updateStatementString);
        if (is_null($this->updateStatement)) {
            throw new Exception("bad statement: '" . $this->updateStatementString . "'");
        }
    }

    /**
     * Gets movies by executing a SQL "select" statement. An empty array
     * is returned if there are no results, or if the query contains an error.
     * 
     * @param String $selectString a valid SQL "select" statement
     * @return array Movie objects
     */
    private function getMoviesByQuery($selectString) {
        $result = [];

        try {
            $stmt = $this->conn->prepare($selectString);
            $stmt->execute();
            $dbresults = $stmt->fetchAll(PDO::FETCH_ASSOC);

            foreach ($dbresults as $r) {
                $movieID = $r['movieID'];
                $title = $r['title'];
                $releaseYear = $r['releaseYear'];
                $length = $r['length'];
                $rating = $r['rating'];
                $obj = new Movie($movieID, $title, $releaseYear, $length, $rating);
                array_push($result, $obj);
            }
        }
        catch (Exception $e) {
            $result = [];
        }
        finally {
            if (!is_null($stmt)) {
                $stmt->closeCursor();
            }
        }
        return $result;
    }

    /**
     * Gets all movies.
     * 
     * @return array Movie objects, possibly empty
     */
    public function getAllMovies() {
        return $this->getMoviesByQuery("select * from MOVIES");
    }

    /**
     * Gets the movie with the specified ID.
     * 
     * @param Integer $id the ID of the movie to retrieve 
     * @return the Movie object with the specified ID, or NULL if not found
     */
    private function getMovieByID($id) {
        $result = NULL;

        try {
            $this->getByIDStatement->bindParam(":movieID", $id);
            $this->getByIDStatement->execute();
            $dbresults = $this->getByIDStatement->fetch(PDO::FETCH_ASSOC); // not fetchAll

            if ($dbresults) {
                $movieID = $dbresults['movieID'];
                $title = $dbresults['title'];
                $releaseYear = $dbresults['releaseYear'];
                $length = $dbresults['length'];
                $rating = $dbresults['rating'];
                $result = new Movie($movieID, $title, $releaseYear, $length, $rating);
            }
        }
        catch (Exception $e) {
            $result = NULL;
        }
        finally {
            if (!is_null($this->getByIDStatement)) {
                $this->getByIDStatement->closeCursor();
            }
        }
        return $result;
    }

    /**
     * Deletes a movie.
     * @param Movie $movie an object whose ID is EQUAL TO the ID of the movie to delete
     * @return boolean indicates whether the movie was deleted
     */
    public function deleteMovie($movie) {
        $success = false;

        $movieID = $movie->getMovieID(); // only the ID is needed

        try {
            $this->deleteStatement->bindParam(":movieID", $movieID);
            $success = $this->deleteStatement->execute(); // this doesn't mean what you think it means
            $rc = $this->deleteStatement->rowCount();
        }
        catch (PDOException $e) {
            $success = false;
        }
        finally {
            if (!is_null($this->deleteStatement)) {
                $this->deleteStatement->closeCursor();
            }
            return $success;
        }
    }

    /**
     * Inserts a movie into the database.
     * 
     * @param Movie $movie an object of type Movie
     * @return boolean indicates if the movie was inserted
     */
    public function insertMovie($movie) {
        $success = false;

        $movieID = $movie->getMovieID();
        $title = $movie->getTitle();
        $releaseYear = $movie->getReleaseYear();
        $length = $movie->getLength();
        $rating = $movie->getRating();

        try {
            $this->insertStatement->bindParam(":movieID", $movieID);
            $this->insertStatement->bindParam(":title", $title);
            $this->insertStatement->bindParam(":releaseYear", $releaseYear);
            $this->insertStatement->bindParam(":length", $length);
            $this->insertStatement->bindParam(":rating", $rating);
            $success = $this->insertStatement->execute();// this doesn't mean what you think it means
        }
        catch (PDOException $e) {
            $success = false;
        }
        finally {
            if (!is_null($this->insertStatement)) {
                $this->insertStatement->closeCursor();
            }
            return $success;
        }
    }

    /**
     * Updates a movie in the database.
     * 
     * @param Movie $movie an object of type Movie, the new values to replace the database's current values
     * @return boolean indicates if the movie was updated
     */
    public function updateMovie($movie) {
        $success = false;

        $movieID = $movie->getMovieID();
        $title = $movie->getTitle();
        $releaseYear = $movie->getReleaseYear();
        $length = $movie->getLength();
        $rating = $movie->getRating();

        try {
            $this->updateStatement->bindParam(":movieID", $movieID);
            $this->updateStatement->bindParam(":title", $title);
            $this->updateStatement->bindParam(":releaseYear", $releaseYear);
            $this->updateStatement->bindParam(":length", $length);
            $this->updateStatement->bindParam(":rating", $rating);
            $success = $this->updateStatement->execute();// this doesn't mean what you think it means
            
        }
        catch (PDOException $e) {
            $success = false;
        }
        finally {
            if (!is_null($this->updateStatement)) {
                $this->updateStatement->closeCursor();
            }
            return $success;
        }
    }

}
// end class movieAccessor

File: JavaMovieDB/src/java/db/MovieAccessor.java
package db;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import entity.Movie;

/**
 * This class provides a central location for accessing the Movie table.
 */
public class MovieAccessor {

    private static Connection conn = null;
    private static PreparedStatement selectAllStatement = null;
    private static PreparedStatement deleteStatement = null;
    private static PreparedStatement insertStatement = null;
    private static PreparedStatement updateStatement = null;

    // constructor is private - no instantiation allowed
    private MovieAccessor() {
    }

    /**
     * Used only by methods in this class to guarantee a database connection.
     *
     * @throws SQLException
     */
    private static void init() throws SQLException {
        if (conn == null) {
            conn = ConnectionManager.getConnection(ConnectionParameters.URL, ConnectionParameters.USERNAME, ConnectionParameters.PASSWORD);
            selectAllStatement = conn.prepareStatement("select * from Movies");
            deleteStatement = conn.prepareStatement("delete from Movies where MOVIEID = ?");
            insertStatement = conn.prepareStatement("insert into Movies values (?,?,?,?,?)");
            updateStatement = conn.prepareStatement("update Movies set TITLE = ?, RELEASEYEAR = ?, LENGTH = ?, RATING = ? where MOVIEID = ?");
        }
    }

    /**
     * Gets all movies.
     *
     * @return a List, possibly empty, of Movie objects
     */
    public static List<Movie> getAllMovies() {
        List<Movie> movies = new ArrayList();
        try {
            init();
            ResultSet rs = selectAllStatement.executeQuery();
            while (rs.next()) {
                int movieID = rs.getInt("MOVIEID");
                String title = rs.getString("TITLE");
                int releaseYear = rs.getInt("RELEASEYEAR");
                int length = rs.getInt("LENGTH");
                int rating = rs.getInt("RATING");
                Movie movie = new Movie(movieID, title, releaseYear, length, rating);
                movies.add(movie);
            }
        } catch (SQLException ex) {
            movies = new ArrayList();
        }
        return movies;
    }

    /**
     * Deletes the Movie with the same ID as the specified movie.
     *
     * @param movie the movie whose ID should be used to match the movie to delete
     * @return <code>true</code> if an movie was deleted; <code>false</code>
     * otherwise
     */
    public static boolean deleteMovie(Movie movie) {
        boolean res;

        try {
            init();
            deleteStatement.setInt(1, movie.getMovieID());
            int rowCount = deleteStatement.executeUpdate();
            res = (rowCount == 1);
        } catch (SQLException ex) {
            res = false;
        }
        return res;
    }

    /**
     * Deletes the Movie with the specified ID.
     *
     * @param id the ID of the movie to delete
     * @return <code>true</code> if an movie was deleted; <code>false</code>
     * otherwise
     */
    public static boolean deleteMovieById(int id) {
        Movie dummy = new Movie(id, "blank", 0, 0, 0);
        return deleteMovie(dummy);
    }
    
    public static boolean insertMovie(Movie movie) {
        boolean res;
        
        try {
            init();
            insertStatement.setInt(1, movie.getMovieID());
            insertStatement.setString(2, movie.getTitle());
            insertStatement.setInt(3, movie.getReleaseYear());
            insertStatement.setInt(4, movie.getLength());
            insertStatement.setInt(5, movie.getRating());
            int rowCount = insertStatement.executeUpdate();
            res = (rowCount == 1);
        }
        catch (SQLException ex) {
            res = false;
        }
        
        return res;
    }

    public static boolean updateMovie(Movie movie) {
        boolean res;
        
        try {
            init();
            updateStatement.setString(1, movie.getTitle());
            updateStatement.setInt(2, movie.getReleaseYear());
            updateStatement.setInt(3, movie.getLength());
            updateStatement.setInt(4, movie.getRating());
            updateStatement.setInt(5, movie.getMovieID());
            int rowCount = updateStatement.executeUpdate();
            res = (rowCount == 1);
        }
        catch (SQLException ex) {
            res = false;
        }
        
        return res;
    }

} // end MovieAccessor
